### DISCUSSION TITLE
How can I set react-number-format so that only the value of the number comes out, not the string with the prefix?

### DISCUSSION THREAD

**Discussion by REDLIGHTFILMS** [react-hook-form/react-hook-form#2511] [Q&A]
My problem is that if I put react-number-format on my text field then the whole string comes out with the commas and I only want to have the value of the number but not the whole string? How can I implement this?

my codesandbox:  https://codesandbox.io/s/react-number-format-kurld

**Comment by bluebill1049** [comment]
1. build a wrapper component around the number input
2. format your data on submit.

one of the reasons I don't like to do at the input level because you will need to format and parse it.

**Reply by REDLIGHTFILMS** [reply]
But if I have 20 fields, in nested objects. I can't call a function 20 times to format the submit. Then how do you do it?

**Reply by bluebill1049** [reply]
I mean format your form values at `submit`.

**Reply by REDLIGHTFILMS** [reply]
I have found this: https://codesandbox.io/s/react-hook-form-normalize-field-er00l 

But how can I give the textfield a label? How can I pass this to the function: MuiCurrencyFormat?

**Reply by bluebill1049** [reply]
take a look the example in the examples folder: https://codesandbox.io/s/react-hook-form-parse-and-format-textarea-furtc?file=/src/index.tsx

**Reply by bluebill1049** [reply]
https://github.com/react-hook-form/react-hook-form/tree/master/examples

**Comment by HadiSDev** [comment]
@REDLIGHTFILMS 

Here you go :D

```
        <Controller name="test" control={control} render={(props) => {
          return <NumberFormat value={props.value} onValueChange={(target)=>{
            props.onChange();
            setValue("test", target.floatValue);
          }} onBlur={props.onBlur}/>
        }}/>
```
